### PR TITLE
[virt_autotest] Collect and upload Guest wickded logs during virtual network test

### PR DESCRIPTION
**Pull request by guoxuguang**
- Background:
Refer to the https://bugzilla.suse.com/show_bug.cgi?id=1177879#c19, there was missed guest wicked configuration and status logs during post_fail_hook to call virt_utils::collect_host_and_guest_logs to collect and upload guest logs.
Required the guest wicked configuration and status logs as below: 
```
#enable wickedd debug logs
perl -i -lpe 's{WICKED_DEBUG=.}{$1=all};s{WICKED_LOG_LEVEL=.}{$1=debug}' /etc/sysconfig/network/config
before wickedd has been started, thus when you change it:
systemctl restart wickedd

# collect the configuration
tar -czf config-ifcfg.tgz /etc/sysconfig/network
wicked show-config > config-dump.log

# collect the status
wicked ifstatus --verbose all > status.log
journalctl -b -o short-precise > journal.log
ip addr show > ip_addr.log
ip route show table all > routes.log
rpm -qa | grep wicked > wicked-version.log
```
Refer to the above the guest wicked configuration and status logs required:
Should be
1. enable guest wicked debugging -> need to implement at libvirt_virtual_network_init step 
2. collect guest wicked configuration and status logs ->guest ssh(recommend)| guest console:: implemented at virt_logs_collector.sh script
3. upload guest wicked configuration and status logs-> guest ssh(recommend)| guest console:: implemented at virt_logs_collector.sh script

NOTE: 
1. confirm that "supportconfig" utility covered the guest (e.g SLE12+ and SLE15+)wicked configuration and status logs as required. So, use with the  "supportconfig" utility  to collect guest (e.g SLE12+ and SLE15+)wicked configuration and status logs. 
2. virt_logs_collector.sh script use with SSH (recommend) way to collect and upload required logs(supportconfig logs) from
guest system to vm host. if vm host found ssh do not work both vm host and guest system together, then used with guest console to collect and upload required logs from guest system to vm host 

So, create this PR to collect and upload guest wicked configuration and status logs during call virt_utils::collect_host_and_guest_logs to collect and upload guest logs. as the log enhancement

- Related ticket: https://progress.opensuse.org/issues/80970
- Verification run:
**OSD:**
[gi-guest_sles15sp3-on-host_sles15sp3-kvm](http://149.44.176.58/tests/5391130)
[gi-guest_sles15sp3-on-host_sles15sp2-kvm](http://149.44.176.58/tests/5393289)
[gi-guest_sles12sp5-on-host_sles15sp3-kvm](http://149.44.176.58/tests/5393301)
[gi-guest_sles15sp2-on-host_sles15sp3-kvm](http://149.44.176.58/tests/5393303)
[gi-guest_sles12sp5-on-host_sles15sp3-xen](http://149.44.176.58/tests/5393302)
[gi-guest_sles15sp3-on-host_sles15sp2-xen](http://149.44.176.58/tests/5404709)
**collect guest wicked log via guest ssh**
[gi-guest_sles15sp3-on-host_sles15sp3-xen](http://149.44.176.58/tests/5391134)
[gi-guest_sles15sp2-on-host_sles15sp3-xen](http://149.44.176.58/tests/5393304)
**collect guest wicked log via guest console**
[gi-guest_sles15sp3-on-host_sles15sp3-kvm](https://github.com/guoxuguang/os-autoinst-distri-opensuse/blob/cc74d84462b5c20137bacfa30e3a21c764b1ad92/virt_logs_collector.logs.guest_console)

### CODE REVIEW

File: lib/virt_feature_test_base.pm [hunk 1]
```diff
@@ -160,7 +160,12 @@ sub post_fail_hook {
     #(caller(0))[3] can help pass calling subroutine name into called subroutine
     $self->junit_log_provision((caller(0))[3]) if get_var("VIRT_AUTOTEST");
 
-    virt_utils::collect_host_and_guest_logs;
+    if (get_var('ENABLE_VIR_NET')) {
```
Comment (waynechen55) [inline lib/virt_feature_test_base.pm:163]: ENABLE_VIR_NET is always enabled in a testsuite if the testsuite has such setting. I think you can use this if appropriate:
my ($package, $filename, $line) = caller;
if $package == xxxxx or $filename == yyyy

File: lib/virt_feature_test_base.pm [hunk 2]
```diff
@@ -160,7 +160,12 @@ sub post_fail_hook {
...
+        virt_utils::collect_host_and_guest_logs('', '', '/tmp/wickded');
+    }
+    else {
+        virt_utils::collect_host_and_guest_logs;
+    }
     upload_logs("/var/log/clean_up_virt_logs.log");
     save_screenshot;
     upload_logs("/var/log/guest_console_monitor.log");
```
Comment (waynechen55) [inline lib/virt_feature_test_base.pm:171]: Please make sure supportconfig does not collect such logs even with "wicked debug logs = enabled". So there is a chance that your further effort can be saved.

Comment (waynechen55) [inline lib/virt_feature_test_base.pm:171]: Is this the only file you are going to modify or there are more files to be modified ? You'd better set WIP if there is ongoing work.

Comment (guoxuguang) [reply lib/virt_feature_test_base.pm:171]: sure, had updated as WIP now

Comment (guoxuguang) [reply lib/virt_feature_test_base.pm:171]: Great point. As wicked developer's mentioned, supportconfig are normal logs -- they may show some error or not. What they show is the config and result at the end (except supportconfig has been called after some manual tweaks/fix attempts, ...). 
So, this PR will collect and upload the related wicked logs from guest system to help wicked developer to debug wicked problem.

**Comment by alice-suse** [comment]
@guoxuguang From PR description, the wicked configuration and status logs are needed by dev. I checked code changes, seems no such explicit code lines for them. How do you plan to have these logs uploaded? BTW verfication run is still missing.

**Comment by guoxuguang** [comment]
> @guoxuguang From PR description, the wicked configuration and status logs are needed by dev. I checked code changes, seems no such explicit code lines for them. How do you plan to have these logs uploaded? BTW verfication run is still missing.

Refer to the wicked configuration and status logs are needed by dev,  I had confirmed them (both SLE12+ and SLE15+ guest system)would be covered from  "supportconfig" utility. 

btw, regarding collecting the guest wicked configuration and status logs via "supportconfig" utility, 
More details from https://progress.opensuse.org/issues/80970?issue_count=12&issue_position=1&next_issue_id=80948#note-5

So,  as my plan for this PR now, go to use with  "supportconfig " utility to collect guest wikced logs (were included the wicked configuration and status logs are needed by dev). 
Also, refer to the current virt_logs_collector.sh script was implemented that "supportconfig -A " utility to collect guest wicked logs via guest console. 
Moreover, refer to the dev''s requested, we need to enable guest wicked debugging at libvirt_virtual_network_init step. 

Thanks for the reminder. I am working on some validation runs for this PR now. After finished them, I will add them here.

File: lib/virt_feature_test_base.pm (hunk 1 above)
Comment (guoxuguang) [reply lib/virt_feature_test_base.pm]: yep, ENABLE_VIR_NET is always enabled in a testsuite if the testsuite has such setting.
had updated code already.

**Review by asmorodskyi — APPROVED** [review]
LGTM

**Comment by guoxuguang** [comment]
Add the final validation run as below:
Verification run:
**OSD:**
[gi-guest_sles15sp3-on-host_sles15sp3-kvm](http://149.44.176.58/tests/5391130)
[gi-guest_sles15sp3-on-host_sles15sp2-kvm](http://149.44.176.58/tests/5393289)
[gi-guest_sles12sp5-on-host_sles15sp3-kvm](http://149.44.176.58/tests/5393301)
[gi-guest_sles12sp5-on-host_sles15sp3-xen](http://149.44.176.58/tests/5393302)
[gi-guest_sles15sp2-on-host_sles15sp3-kvm](http://149.44.176.58/tests/5393303)
**collect guest wicked log via guest ssh**
[gi-guest_sles15sp3-on-host_sles15sp2-xen](http://149.44.176.58/tests/5393290)
[gi-guest_sles15sp3-on-host_sles15sp3-xen](http://149.44.176.58/tests/5391134)
[gi-guest_sles15sp2-on-host_sles15sp3-xen](http://149.44.176.58/tests/5393304)
**collect guest wicked log via guest console**
[gi-guest_sles15sp3-on-host_sles15sp3-kvm](https://github.com/guoxuguang/os-autoinst-distri-opensuse/blob/cc74d84462b5c20137bacfa30e3a21c764b1ad92/virt_logs_collector.logs.guest_console)

@alice-suse @Julie-CAO @waynechen55 @nanzhang2020 please help me review this PR.

**Comment by guoxuguang** [comment]
> LGTM

**Comment by alice-suse** [comment]
> Add the final validation run as below:
> Verification run:
> **OSD:**
> [gi-guest_sles15sp3-on-host_sles15sp3-kvm](http://149.44.176.58/tests/5391130)
> [gi-guest_sles15sp3-on-host_sles15sp2-kvm](http://149.44.176.58/tests/5393289)
> [gi-guest_sles12sp5-on-host_sles15sp3-kvm](http://149.44.176.58/tests/5393301)
> [gi-guest_sles12sp5-on-host_sles15sp3-xen](http://149.44.176.58/tests/5393302)
> [gi-guest_sles15sp2-on-host_sles15sp3-kvm](http://149.44.176.58/tests/5393303)
> **collect guest wicked log via guest ssh**
> [gi-guest_sles15sp3-on-host_sles15sp2-xen](http://149.44.176.58/tests/5393290)

There is no virt_logs_all.tar.gz. Why?

> [gi-guest_sles15sp3-on-host_sles15sp3-xen](http://149.44.176.58/tests/5391134)
> [gi-guest_sles15sp2-on-host_sles15sp3-xen](http://149.44.176.58/tests/5393304)
> **collect guest wicked log via guest console**
> [gi-guest_sles15sp3-on-host_sles15sp3-kvm](https://github.com/guoxuguang/os-autoinst-distri-opensuse/blob/cc74d84462b5c20137bacfa30e3a21c764b1ad92/virt_logs_collector.logs.guest_console)

This link is not a job link.

> @alice-suse @Julie-CAO @waynechen55 @nanzhang2020 please help me review this PR.

**Comment by guoxuguang** [comment]
> > Add the final validation run as below:
> > Verification run:
> > **OSD:**
> > [gi-guest_sles15sp3-on-host_sles15sp3-kvm](http://149.44.176.58/tests/5391130)
> > [gi-guest_sles15sp3-on-host_sles15sp2-kvm](http://149.44.176.58/tests/5393289)
> > [gi-guest_sles12sp5-on-host_sles15sp3-kvm](http://149.44.176.58/tests/5393301)
> > [gi-guest_sles12sp5-on-host_sles15sp3-xen](http://149.44.176.58/tests/5393302)
> > [gi-guest_sles15sp2-on-host_sles15sp3-kvm](http://149.44.176.58/tests/5393303)
> > **collect guest wicked log via guest ssh**
> > [gi-guest_sles15sp3-on-host_sles15sp2-xen](http://149.44.176.58/tests/5393290)
> 
> There is no virt_logs_all.tar.gz. Why?
Opps, let me check with it. 
> 
> > [gi-guest_sles15sp3-on-host_sles15sp3-xen](http://149.44.176.58/tests/5391134)
> > [gi-guest_sles15sp2-on-host_sles15sp3-xen](http://149.44.176.58/tests/5393304)
> > **collect guest wicked log via guest console**
> > [gi-guest_sles15sp3-on-host_sles15sp3-kvm](https://github.com/guoxuguang/os-autoinst-distri-opensuse/blob/cc74d84462b5c20137bacfa30e3a21c764b1ad92/virt_logs_collector.logs.guest_console)
> 
> This link is not a job link.
good question,  for now,  I can't let guest network broken on OSD environment. So, I have to manually trigger virt_logs_collector.sh on my test environment to collect guest wicked log via guest console so far. 

Hope that can answer your question here. 

FYI
Why do not use with - DHCPDECLINE IP address abandoned on OSD environment to confirm collect and upload guest wicked
log via guest console way?
From the latest OSD SLE build136.1, after disabled all workaround for DHCPDECLINE IP address abandoned. still can't reproduce  this DHCPDECLINE IP address abandoned on OSD environment any more to let guest network broken -> use with guest console to collect and upload guest wicked logs. 
So, I have to use with virt_logs_collector.sh directly to collect guest wicked logs on my testing environment. 

btw, this is DHCPDECLINE IP address abandoned on OSD environment was the other question.

**Comment by guoxuguang** [comment]
> > [gi-guest_sles15sp3-on-host_sles15sp2-xen](http://149.44.176.58/tests/5393290) 
> There is no virt_logs_all.tar.gz. Why?

@alice-suse  Quick update, 
[gi-guest_sles15sp3-on-host_sles15sp2-xen](http://149.44.176.58/tests/5393290) , should be the nvalid validation run for this PR. 

But, after basic analyze again. it would be the ad-hoc failure for [gi-guest_sles15sp3-on-host_sles15sp2-xen](http://149.44.176.58/tests/5393290). So, I had created a process ticket to trace this failure
https://progress.opensuse.org/issues/88435. 

Refer to the latest validation run for [gi-guest_sles15sp3-on-host_sles15sp2-xen](http://149.44.176.58/tests/5404709) , results looks good without any error.

**Pull request merged**